### PR TITLE
GetValidatorPerformance http endpoint

### PR DESCRIPTION
**Pull request by saolyn**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**
This PR adds the http endpoint for `GetValidatorPerformance`.

**Which issues(s) does this PR fix?**
N/A

**Other notes for review**

### CODE REVIEW

**Review by rkapka — CHANGES_REQUESTED** [review]
The whole handler implementation is duplication of code from v1alpha1's `GetValidatorPerformance`. We should avoid this.

File: beacon-chain/rpc/prysm/validator/validator_performance.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	BalancesBeforeEpochTransition []uint64 `json:"balances_before_epoch_transition,omitempty"`
+	BalancesAfterEpochTransition  []uint64 `json:"balances_after_epoch_transition,omitempty"`
+	MissingValidators             [][]byte `json:"missing_validators,omitempty"`
+	InactivityScores              []uint64 `json:"inactivity_scores,omitempty"`
+}
+
+// GetValidatorPerformance is an HTTP handler for GetValidatorPerformance.
+func (vs *Server) GetValidatorPerformance(w http.ResponseWriter, r *http.Request) {
+	if vs.SyncChecker.Syncing() {
+		handleHTTPError(w, "Syncing", http.StatusInternalServerError)
```
Comment (rkapka) [inline beacon-chain/rpc/prysm/validator/validator_performance.go:33]:
```suggestion
		handleHTTPError(w, "Syncing", http.StatusInternalServerError)
```
Syncing should return 503 status code

File: beacon-chain/rpc/core/errors.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	switch reason {
+	case Internal:
+		return codes.Internal
+	case Unavailable:
+		return codes.Unavailable
+	case BadRequest:
+		return codes.InvalidArgument
+	// Add more cases for other error reasons as needed
+	default:
+		return codes.Unknown
```
Comment (rkapka) [inline beacon-chain/rpc/core/errors.go:33]: For HTTP we return `StatusInternalServerError` by default, so here it should be `codes.Internal`

Comment (saolyn) [reply beacon-chain/rpc/core/errors.go]: Done

File: beacon-chain/rpc/prysm/v1alpha1/beacon/validators.go [hunk 3]
```diff
@@ -659,153 +660,14 @@ func (bs *Server) GetValidatorPerformance(
 	ctx context.Context, req *ethpb.ValidatorPerformanceRequest,
 ) (*ethpb.ValidatorPerformanceResponse, error) {
 	if bs.SyncChecker.Syncing() {
-		return nil, status.Errorf(codes.Unavailable, "Syncing to latest head, not ready to respond")
-	}
-
-	headState, err := bs.HeadFetcher.HeadState(ctx)
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "Could not get head state: %v", err)
+		return nil, status.Error(core.ErrorReasonToGRPC(core.Unavailable), "syncing to latest head, not ready to respond")
```
Comment (rkapka) [inline beacon-chain/rpc/prysm/v1alpha1/beacon/validators.go:663]: Here you can use `codes.Unavailable` directly, without calling the helper function

**Review by rkapka — APPROVED** [review]

**Review by rkapka — APPROVED** [review]

**Review by rkapka — APPROVED** [review]

**Pull request merged**